### DISCUSSION TITLE
Can we create new folder AND move pre-selected folders/files into it at the same time?

### DISCUSSION THREAD

**Discussion by sangdth** [nvim-tree/nvim-tree.lua#1232] [Ideas]
I saw my teammate has an editor, that allow him to refactor his folders/files.

The steps:
- He selects folders, and files. Says `first/first-file.ts`, `second/second-file.ts`, `third/` folder.
- Right click, choose "Refactor"
- Type the new folder name "`new-place`" then confirm the action.
- The result: All selected folders and files are moved into the newly created folder. They will be `new-place/first-file.ts`, `new-place/second-file.ts`, `new-place/third`

Do you think it's possible for us to archive the similar experience with nvim-tree or at least with neovim?

Thanks.

**Comment by kyazdani42** [comment]
Hi, it's not feasible at the moment, but i think this would make a great addition :) Please raise a new feature request for this !

**Reply by sangdth** [reply]
Thank you! I created it here #1247 
Btw if you could provide some guides, I would love to provide some helps. I know some basic lua knowledge and huge love with neovim community.

**Reply by kyazdani42** [reply]
:+1: i should extend the `contributing.md` a bit, it's too simple for newcomers ^^ I'll see when i have some time to do so

**Reply by alex-courtis** [reply]
There is some helpful information here: [CONTRIBUTING.md](https://github.com/kyazdani42/nvim-tree.lua/blob/master/CONTRIBUTING.md)

My basic dev setup:
1. Fork this repo
2. Create /tmp/nvt-min.lua as per [Bug Report](https://github.com/kyazdani42/nvim-tree.lua/issues/new/choose)
3. Change the plugin location to your local repo e.g. `"~/src/nvim-tree.lua"`
4. `nvim -nu /tmp/nvt-min.lua`

[Logging](https://github.com/kyazdani42/nvim-tree.lua#diagnostic-logging) is useful when developing. Create a new type or set `all = true`.

**Reply by sangdth** [reply]
@alex-courtis really useful comment, I will try soon
